website/photos/services.py

```diff
@@ -70,3 +84,99 @@ def get_annotated_accessible_albums(request, albums):
         )
 
     return albums
+
+
+def extract_archive(request, album, archive):
```

Comment:
Wouldn't it be easier to extract the file into a directory and then iterate through that directory? Seems like it may be less error prone. Also I would suggest using the `mimetypes` module to check if the file is a zip/tar file.

This method would also remove rechecking for zipfile/tarfile below

Reply:
Probably true, but same story as the other comment: just moving around. I really bumped into problems that the way this worked was all getting into quite large files, so that is why I refactored this into services.py before fixing the bug that I was actually working on.

I'd love to improve this code, but it's not getting any worse than what we had before. And since this has been open for 2 weeks I'd like to move those optimalisations to a new issue